drivers/filter_wheel/ioptron_wheel.cpp

```diff
@@ -0,0 +1,310 @@
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Library General Public License for more details.
+
+ You should have received a copy of the GNU Library General Public License
+ along with this library; see the file COPYING.LIB.  If not, write to
+ the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ Boston, MA 02110-1301, USA.
+*******************************************************************************/
+
+#include "ioptron_wheel.h"
+
+#include "indicom.h"
+#include <connectionplugins/connectionserial.h>
+#include <cstring>
+#include <memory>
+#include <termios.h>
+
+#define iEFW_TIMEOUT 4
+
+/*
+command :
+:WMnn#          Response:  1
+get position ，nn filter num。
+
+:WP#            Response:  nn#  or  -1#
+get curr position,nn =curr position，if is moving，return -1。
+
+:WOnnsnnnnn#	Response:  1
+set offset。nn =filter num , snnnnn=offset
+
+:WFnn#          Response:     snnnnn#
+get offset 。nn = filter num , snnnnn=offset
+
+“:DeviceInfo#”		Response: “nnnnnnnnnnnn#”
+This command includes 12 digits.
+The 7th and 8th digit means model of filter wheel.
+Model number  99 is iEFW-15, 98 is iEFW18
+
+:FW1#			Response:  nnnnnnnnnnnn#
+get firmware version
+
+
+*/
+
+
+static std::unique_ptr<iEFW> iefw_wheel(new iEFW());
+
+iEFW::iEFW()
+{
+    setVersion(1, 0);
+    setFilterConnection(CONNECTION_SERIAL);
+}
+
+const char *iEFW::getDefaultName()
+{
+    return "iOptron Wheel";
+}
+
+bool iEFW::initProperties()
+{
+    INDI::FilterWheel::initProperties();
+//    serialConnection->setDefaultPort("/dev/ttyUSB3");
+    serialConnection->setDefaultBaudRate(Connection::Serial::B_115200);
+    FilterSlotN[0].min = 1;
+    FilterSlotN[0].max = 8;
+    FirmwareTP[0].fill("FIRMWARE", "Firmware", "240101240101");
+    FirmwareTP.fill(getDeviceName(), "FIRMWARE_ID", "iEFW Firmware", FILTER_TAB, IP_RO, 60, IPS_IDLE);
+    WheelIDTP[0].fill("MODEL", "Model", "iEFW");
+    WheelIDTP.fill(getDeviceName(), "MODEL_ID", "iEFW Model", FILTER_TAB, IP_RO, 60, IPS_IDLE);
+    return true;
+}
+
+bool iEFW::updateProperties()
+{
+    INDI::FilterWheel::updateProperties();
+
+    if (isConnected())
+      {
+        defineProperty(HomeSP);
+        defineProperty(FirmwareTP);
+        defineProperty(WheelIDTP);
+	bool rc1=getiEFWInfo();
+        bool rc2=getiEFWfirmwareInfo();
+        getFilterPos();
+	return (rc1 && rc2);
+      }
+    else
+      {
+        deleteProperty(HomeSP);
+        deleteProperty(FirmwareTP);
+        deleteProperty(WheelIDTP);
+      }
+    return true;
+}
+
+bool iEFW::ISNewSwitch(const char *dev, const char *name, ISState *states, char *names[], int n)
+{
+    if (dev != nullptr && strcmp(dev, getDeviceName()) == 0)
+    {
+        if (HomeSP.isNameMatch(name))
+        {
+            if (getiEFWID())
+            {
+                LOG_INFO("Filter getid.");
```

Comment:
Perhaps a better message should displayed to the user?

LOG_INFO("Filter is at home position.");

for example?